### PR TITLE
Add contact support functionality #53

### LINKED ISSUE #53: #21: Add contact support functionality
1) Read guideline [here](https://gist.github.com/iamnovichek/801ac395a06a7d8b04b022084ace684d)
2) Add Django model `Support` with following fields: `user`,  `telegram_username` and update migrations
3) Create a function `get_support_link` in `tracker/utils.py` creates and returns a clickable link to support's DM
4) Add telegram button `Contact support` that provides a clickable link to support's DM for each repository
5) Provide some screenshots to prove that your feature works as expected

### PR DESCRIPTION
**Pull request by 0xNgoo**
should Closes #53

### CODE REVIEW

**Comment by 0xNgoo** [comment]
@iamnovichek Please where do i create the Add telegram button `Contact support`

File: tracker/migrations/0004_support.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Generated by Django 5.1.3 on 2024-12-01 11:16
+
+import django.db.models.deletion
```
Comment (iamnovichek) [inline tracker/migrations/0004_support.py]: Pls sync with main
image

File: tracker/models.py [hunk 2]
```diff
@@ -278,4 +278,23 @@ def create_telegram_user(sender, instance, created, **kwargs):
...
+
+    user = models.ForeignKey(CustomUser, on_delete=models.CASCADE)
+    telegram_username = models.CharField(max_length=DefaultModelValues.name_max_length)
+
+    def __str__(self) -> str:
+        """
+        Returns a string representation of the Support instance.
+        :return: str
+        """
+        return f"Support request from {self.telegram_username}"
```
Comment (iamnovichek) [inline tracker/models.py]: No, the idea is that all support request process will be through Telegram. So pls change text.

**Review by iamnovichek — CHANGES_REQUESTED** [review]

**Comment by iamnovichek** [comment]
> @iamnovichek Please where do i create the Add telegram button `Contact support`

@0xNgoo inside `tracker/telegram/bot.py`.

**Comment by 0xNgoo** [comment]
@iamnovichek, I don't have any idea how i can test and get a screenshot for this

File: tracker/utils.py [hunk 3]
```diff
@@ -374,3 +374,34 @@ def get_time_before_deadline(issue: dict) -> str:
...
+    """
+    Gets the support contact for a specific repository.
+    
+    :param author: Repository author
+    :param repo_name: Repository name
+    :return: Support instance or None
+    """
+    from .models import Repository, Support
+    try:
+        repository = Repository.objects.get(author=author, name=repo_name)
```
Comment (iamnovichek) [inline tracker/utils.py]: Pls use `filter().first()` here as well to avoid `try/except` statement

**Review by iamnovichek — CHANGES_REQUESTED** [review]

**Comment by iamnovichek** [comment]
> @iamnovichek, I don't have any idea how i can test and get a screenshot for this

But how did you write your code without testing it?

**Comment by iamnovichek** [comment]
@0xNgoo Pls sync with main

**Comment by iamnovichek** [comment]
> @iamnovichek, I don't have any idea how i can test and get a screenshot for this

@0xNgoo 

1) Set DEBUG = True in `settings.py`
2) Start your project (check `SETUP.md`) 
3) Sign up (choose team-lead role)
4) Add at least one repo
5) Now you should be able to test your code.
You can find a demo video link in `README.md`

**Comment by 0xNgoo** [comment]
WhatsApp Image 2024-12-01 at 2 36 40 PM

**Comment by iamnovichek** [comment]
@0xNgoo pls open your PR so I can merge it

**Pull request merged**